### PR TITLE
#95/chore/check new linter ruff

### LINKED ISSUE #95: Check new linter Ruff
## Why

Ruff is new linter and it is very fast. It worth to check detail. (Potentially, it might be better than flake8.)
Actually some tech company, start to use Ruff instead of flake8 or isort: https://github.com/astral-sh/ruff.

## Definition of Done

Basic info about Ruff is summarized and it is possible to start discussion about how good it is.

## How

- Read info about Ruff and compare features with flake8 and isort.
- Install and try ruff (apply both and check if no difference).

### PR DESCRIPTION
**Pull request by YoshikiKubotani**
## Issue URL

Close: #95 

## Change overview

- Add the new linter `ruff` written in Rust, and remove `flake8` and `isort` from dev dependencies
- Update the python libraries listed on dev dependencies

[Add changes below on 2024/04/27]
- Replace `black` with [Ruff formatter](https://docs.astral.sh/ruff/formatter/) (released as of v0.1.2)
- Add the recommended Ruff linter rules compatible with Ruff formatter [[link](https://docs.astral.sh/ruff/formatter/#conflicting-lint-rules)]
- Remove `.flake8` file
- Describe how to loosen the rules of Ruff in `README.md`
- Update `README.md` representation

## How to test

- ~~Make sure to work `make ruff`, `make ruff-check` under the core container~~
- Make sure to run `make format` with no warnings under the core container

## Note for reviewers

N/A

### CODE REVIEW

**Review by gatheluck — APPROVED** [review]
LGTM!!

**Comment by gatheluck** [comment]
@YoshikiKubotani 
CI is raising error. Can you fix it?

File: README.md [hunk 1]
```diff
@@ -4,54 +4,51 @@
...
+    ├── outputs/               <- Outputs.
     │
-    ├── src/               <- Source code. This sould be Python module.
+    ├── src/                   <- Python Source code.
     │
-    ├── tests/             <- Test codes.
+    ├── tests/                 <- Test code.
     │
-    ├── .flake8            <- Setting file for Flake8.
+    ├── .flake8                <- Flake8 configuration.
```
Comment (gatheluck) [inline README.md]: We will remove `.flake8` file in this PR, right?

**Review by gatheluck — APPROVED** [review]
I added one commend about README.md, except that part LGTM!

File: README.md (hunk 1 above)
Comment (YoshikiKubotani) [reply README.md]: Thank you for your suggestion.
This is so true.

**Pull request merged**